### PR TITLE
Style Guide - Foundations/JavaScript Basics 2: Assignments

### LINKED ISSUE #24242: Style Guide - Foundations/JavaScript Basics 2: Assignments

### PR DESCRIPTION
**Pull request by matristle**
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

Closes #24242

**2. This PR:**
* Encapsulates numbered lists in Assignments with div tags and with proper attributes (lacking in Clean Code)
* Transforms Practice sections to Assignment sections (Fundamentals Part 4, DOM Manipulation and Events, Fundamentals Part 5)
* Extracts Optional/Extra Credit items from lists in the Assignment section to dedicated Extra Credit Sections (Project: Etch-a-Sketch, Project: Calculator) 

**3. Additional Information:**

There are some added spaces that I'm not sure if they meet the last bullet point from Description of Issue.

### CODE REVIEW

File: foundations/javascript_basics/clean_code.md [hunk 1]
```diff
@@ -89,8 +89,11 @@ There are many different opinions on what constitutes great JavaScript code.  Th
 
 Read through these articles that discuss a few elements of writing good clean code.
```
Comment (thatblindgeye) [inline foundations/javascript_basics/clean_code.md:90]: We can remove this line.

File: foundations/javascript_basics/fundamentals-5.md [hunk 2]
```diff
@@ -21,18 +21,18 @@ Besides being a quick and handy way to store data, arrays also have a set of fun
 1. Start out by watching [this video](https://www.youtube.com/watch?v=HB1ZC7czKRs) from Wes Bos.  To follow along, use your local Javascript30 repository. If you don't have the repo yet, clone it from [here](https://github.com/wesbos/JavaScript30).
 2. Watch and code with [Part 2](https://www.youtube.com/watch?v=QNmRfyNg1lw) of Wes Bos's array series.
 
-### Practice
+### Assignment
 
 You have been through a _ton_ of content, and it would not be surprising if you were feeling rather overwhelmed at this point.  Let's take a little time to slow down and practice this stuff with more coding exercises.  I'll warn you now: some of these get pretty difficult, but if you've done the lessons you should have covered everything you need to know to solve them.  Don't be afraid to go back and review if you find that you can't remember something, that's part of the process.  If you get stuck come join us in the chatroom!
```
Comment (thatblindgeye) [inline foundations/javascript_basics/fundamentals-5.md:26]: I think this line can also be removed

File: foundations/javascript_basics/fundamentals-5.md [hunk 3]
```diff
@@ -21,18 +21,18 @@ Besides being a quick and handy way to store data, arrays also have a set of fun
...
-- fibonacci
-- getTheTitles
-- findTheOldest
+* calculator
+* palindromes
+* fibonacci
+* getTheTitles
+* findTheOldest
 
 Don't remember how to setup and test the exercises? No worries! Check out the [Fundamentals Part 4 lesson](https://www.theodinproject.com/lessons/foundations-fundamentals-part-4#practice) for a refresher.
```
Comment (thatblindgeye) [inline foundations/javascript_basics/fundamentals-5.md:37]: For this line, let's add in at the end something from the line above being removed:

```suggestion
Don't remember how to setup and test the exercises? No worries! Check out the [Fundamentals Part 4 lesson](https://www.theodinproject.com/lessons/foundations-fundamentals-part-4#practice) for a refresher. If you feel yourself getting overwhelmed or stuck, don't be afraid to go back and review or ask for help on our Discord!
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Awesome job on this! Had a few change requests below 🙂

File: foundations/javascript_basics/clean_code.md (hunk 1 above)
Comment (matristle) [reply foundations/javascript_basics/clean_code.md]: Done. https://github.com/TheOdinProject/curriculum/pull/24248/commits/e93fbe271d6412ecffbddb156adc3d656b93b9e2

File: foundations/javascript_basics/fundamentals-5.md (hunk 2 above)
Comment (matristle) [reply foundations/javascript_basics/fundamentals-5.md]: Done. https://github.com/TheOdinProject/curriculum/pull/24248/commits/8e620c89862d505387b7aa7f5c9439a8302cd2f1

Although formatting consistency was prompted here for sure, I wasn't certain if I was ditching an important part or not.

File: foundations/javascript_basics/fundamentals-5.md (hunk 3 above)
Comment (matristle) [reply foundations/javascript_basics/fundamentals-5.md]: Done. https://github.com/TheOdinProject/curriculum/pull/24248/commits/27c91f1cef63202ae0ab7c0f79913ab0a8b19841

**Pull request merged**